tests/Fixtures/TestBundle/Document/RelatedToUrlEncodedId.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\Document;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * Related To URL Encoded ID represent an association for encoding special characters
+ * TODO: We should not really need to have urlEncodedIdResource should we? Determine that the related resource has this requirement and then apply it as a requirement here would be preferable. Not sure about composite IDs though.
```

Comment:
Maybe if you use a `.` somewhere as symfony has a default regex for `.{_format}` IIRC

Reply:
That's right, again part of the 'other issue' I mentioned. Just left this to do in there. But API Platform supports modifying the route parameter requirement on a resource so thought it should naturally cascade to resources that have a relation to it like here instead of this manual configuration.